lib/node_modules/@stdlib/math/base/special/logaddexp/test/test.js

```diff
@@ -0,0 +1,113 @@
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var tape = require( 'tape' );
+var isnan = require( '@stdlib/math/base/assert/is-nan' );
+var abs = require( '@stdlib/math/base/special/abs' );
+var PINF = require( '@stdlib/constants/math/float64-pinf' );
+var NINF = require( '@stdlib/constants/math/float64-ninf' );
+var EPS = require( '@stdlib/constants/math/float64-eps' );
+var randu = require( '@stdlib/random/base/randu' );
+var logaddexp = require( './../lib' );
+
+
+// FIXTURES //
+
+var data = require( './fixtures/python/data.json' );
+
+
+// TESTS //
+
+tape( 'main export is a function', function test( t ) {
+	t.ok( true, __filename );
+	t.equal( typeof logaddexp, 'function', 'main export is a function' );
+	t.end();
+});
+
+tape( 'the function returns `NaN` if provided `NaN`', function test( t ) {
+	var v;
+
+	v = logaddexp( NaN, 1 );
+	t.equal( isnan( v ), true, 'returns NaN' );
+	v = logaddexp( 1, NaN );
+	t.equal( isnan( v ), true, 'returns NaN' );
+	t.end();
+});
+
+tape( 'the function returns `+infinity` if provided with `+infinity` and `x`', function test( t ) {
+	var iterations;
+	var x;
+	var v;
+	var i;
+
+	iterations = 1000;
+	for ( i = 0; i < iterations; i++ ) {
+		x = ( randu() * 1000.0 ) - 500.0;
+		v = logaddexp( PINF, x );
+		t.strictEqual( v, PINF, 'returns +infinity when provided +infinity and '+x+'.' );
+		v = logaddexp( x, PINF );
+		t.strictEqual( v, PINF, 'returns +infinity when provided +infinity and '+x+'.' );
+	}
+	t.end();
+});
+
+tape( 'the function returns `x` if provided with `-infinity` and `x`', function test( t ) {
+	var iterations;
+	var x;
+	var v;
+	var i;
+
+	iterations = 1000;
+	for ( i = 0; i < iterations; i++ ) {
+		x = ( randu() * 1000.0 ) - 500.0;
+		v = logaddexp( NINF, x );
+		t.strictEqual( v, x, 'returns '+x+' when provided -infinity and '+x+'.' );
+		v = logaddexp( x, NINF );
+		t.strictEqual( v, x, 'returns '+x+' when provided -infinity and '+x+'.' );
+	}
+	t.end();
+});
+
+tape( 'the function accurately computes the natural logarithm of exp(x) + exp(y)', function test( t ) {
+	var delta;
+	var tol;
+	var ex;
+	var i;
+	var v;
+	var y;
+	var x;
+
+	for ( i = 0; i < data.length; i++ ) {
+		x = data[i].input[0];
+		y = data[i].input[1];
+		ex = data[i].expected;
+		v = logaddexp( x, y );
+		if ( v === ex ) {
+			t.strictEqual( v, ex, 'returns '+v+' when provided '+x+' and '+y+'.' );
+		} else {
+			delta = abs( ex - v );
+			tol = EPS * abs( ex );
+			t.ok( delta <= tol, 'within tolerance. returns '+v+' when provided '+x+' and '+y+'. expected: '+ex+', Δ: '+delta+'.');
```

Comment:
Common practice is to also include the `tol` in the assertion description. This allows a developer to see, from the test output, how different the actual delta is from the "allowed" delta (aka tolerance).